src/server/game/Battlegrounds/BattlegroundMgr.cpp

```diff
@@ -794,14 +794,30 @@ uint8 BattlegroundMgr::BGArenaType(BattlegroundQueueTypeId bgQueueTypeId)
 
 void BattlegroundMgr::ToggleTesting()
 {
-    m_Testing = !m_Testing;
-    sWorld->SendWorldText(m_Testing ? LANG_DEBUG_BG_ON : LANG_DEBUG_BG_OFF);
+    if (sWorld->getBoolConfig(CONFIG_DEBUG_BATTLEGROUND))
+    {
+        m_Testing = true;
+        sWorld->SendWorldText(LANG_DEBUG_BG_CONF);
+    }
+    else
+    {
+        m_Testing = !m_Testing;
+        sWorld->SendWorldText(m_Testing ? LANG_DEBUG_BG_ON : LANG_DEBUG_BG_OFF);
+    }
 }
 
 void BattlegroundMgr::ToggleArenaTesting()
 {
-    m_ArenaTesting = !m_ArenaTesting;
-    sWorld->SendWorldText(m_ArenaTesting ? LANG_DEBUG_ARENA_ON : LANG_DEBUG_ARENA_OFF);
+    if (sWorld->getBoolConfig(CONFIG_DEBUG_ARENA))
+    {
+        m_ArenaTesting = true;
+        sWorld->SendWorldText(LANG_DEBUG_ARENA_CONF);
+    }
+    else
+    {
+        m_ArenaTesting = !m_ArenaTesting;
+        sWorld->SendWorldText(m_ArenaTesting ? LANG_DEBUG_ARENA_ON : LANG_DEBUG_ARENA_OFF);
+    }
```

Comment:
You could also write this code in this way to make little modifications.
```C++
    m_ArenaTesting = (sWorld->getBoolConfig(CONFIG_DEBUG_ARENA) || !m_ArenaTesting);
    sWorld->SendWorldText((m_ArenaTesting || sWorld->getBoolConfig(CONFIG_DEBUG_ARENA)) ? LANG_DEBUG_ARENA_ON : LANG_DEBUG_ARENA_OFF);
```

but it's the same in the end, personally I don't know what is best to the readability.

Reply:
Personally I prefer the way I did it since I think it gives a more clear view of what is going on. But I am up for suggestions 🍏